### PR TITLE
Remove PetStore

### PR DESCRIPTION
**Pull request by mcgallan**
# Description
<i>Add your description here!</i>

# Checklist

To ensure a quick review and merge, please ensure:
- [ ] The PR has a understandable title and description explaining the _why_ and _what_.
- [ ] The PR is opened in draft if not ready for review yet.
   - If opened in draft, please allocate sufficient time (24 hours) after moving out of draft for review
- [ ] The branch is recent enough to not have merge conflicts upon creation.

# Ready to Land?
- [ ] Build is completely green
   - Submissions with test failures require tracking issue and approval of a CODEOWNER
- [ ] At least one +1 review by a CODEOWNER
- [ ] All -1 reviews are confirmed resolved by the reviewer 
   - Override/Marking reviews stale must be discussed with CODEOWNERS first

### CODE REVIEW

**Review by ArcturusZhang — APPROVED** [review]

**Review by pshao25 — CHANGES_REQUESTED** [review]
Please do the same fix as FirstTest.

File: test/TestProjects/PetStore-TypeSpec/PetStore-TypeSpec.tsp [hunk 1]
```diff
@@ -99,32 +99,6 @@ model Tuna extends Fish {
 @summary("Manage your pets.")
 @route("/pets")
 namespace Pets {
-  @doc("Delete a pet.")
```
Comment (pshao25) [inline test/TestProjects/PetStore-TypeSpec/PetStore-TypeSpec.tsp:102]: Models need to be removed as well.

File: test/TestProjects/PetStore-TypeSpec/PetStore-TypeSpec.tsp [hunk 2]
```diff
@@ -99,32 +99,6 @@ model Tuna extends Fish {
...
-  @convenientAPI(true)
-  @get op getPetByKind(@path kind: PetKind): Pet | Error;
-
-  @convenientAPI(true)
-  @get op GetFirstPet(@query start?: int32): Pet | Error;
-
-  @convenientAPI(true)
-  @route("/getFish")
-  @get op getFish(@query kind?: string): Fish | Error;
-
```
Comment (pshao25) [inline test/TestProjects/PetStore-TypeSpec/PetStore-TypeSpec.tsp:127]: Ask @archerzz if we still need LRO tests in the autorest.csharp. If not, we could remove the whole file.

File: test/TestProjects/PetStore-TypeSpec/PetStore-TypeSpec.tsp (hunk 1 above)
Comment (mcgallan) [reply test/TestProjects/PetStore-TypeSpec/PetStore-TypeSpec.tsp:102]: fixed

File: test/TestProjects/PetStore-TypeSpec/PetStore-TypeSpec.tsp (hunk 2 above)
Comment (archerzz) [reply test/TestProjects/PetStore-TypeSpec/PetStore-TypeSpec.tsp:127]: > Ask @archerzz if we still need LRO tests in the autorest.csharp. If not, we could remove the whole file.

Let's focus on `cadl-ranch` LRO tests for now. We should guide green field SDKs to use standard template as much as possible.

Comment (mcgallan) [reply test/TestProjects/PetStore-TypeSpec/PetStore-TypeSpec.tsp:127]: fixed

**Review by pshao25 — APPROVED** [review]

**Pull request merged**